src/ast.rs

```diff
@@ -72,50 +73,47 @@ impl From<((usize, usize, usize), (usize, usize, usize))> for Span {
 }
 
 /// Represents a name of an entity, such as a type, variable, function, etc.
-#[derive(Debug, PartialEq)]
-pub struct RawIdentifier<'a> {
-    pub name: &'a str,
-}
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawIdentifier(pub String);
 
-pub type Identifier<'a> = Spanned<RawIdentifier<'a>>;
+pub type Identifier = Spanned<RawIdentifier>;
 
-impl<'a> From<ParseInput<'a>> for RawIdentifier<'a> {
+impl<'a> From<ParseInput<'a>> for RawIdentifier {
     fn from(span: ParseInput<'a>) -> Self {
-        RawIdentifier {
-            name: span.fragment(),
-        }
+        RawIdentifier((*span.fragment()).to_owned())
     }
 }
 
-/// A parameter to a function, e.g., `foo: MyType`.
-#[derive(Debug, PartialEq)]
-pub struct RawParameter<'a> {
-    pub name: Identifier<'a>,
-    pub param_type: Identifier<'a>,
+/// Introduces a variable with its name and type, e.g., `foo : MyType`. E.g. used in function
+/// type signature parameter lists, `let` bindings, etc.
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawVariableDeclaration {
+    pub name: Identifier,
+    pub typ: Identifier,
 }
 
-pub type Parameter<'a> = Spanned<RawParameter<'a>>;
-pub type ParameterList<'a> = Spanned<Vec<Parameter<'a>>>;
+pub type VariableDeclaration = Spanned<RawVariableDeclaration>;
```

Comment:
I think we can use "Decl" instead of "Declaration" throughout; it's a pretty std abbr.

Reply:
Ok. I've been following google C++ style guide

> Do not worry about saving horizontal space as it is far more important to make your code immediately understandable by a new reader. Minimize the use of abbreviations that would likely be unknown to someone outside your project (especially acronyms and initialisms). Do not abbreviate by deleting letters within a word. As a rule of thumb, an abbreviation is probably OK if it's listed in Wikipedia

"decl" isn't found in wikipedia, but I have seen it used in other programming contexts before, so I'm fine with using it. That said, I err on the side of clarity over abbreviation.

> std abbr.

haha :)

Reply:
done